### ISSUE TITLE
Adding quality stats per link and per path

### ISSUE THREAD

**Issue by leonardomaccari** [interop-dev/django-netjsongraph#2]
I'd like to add some stats of the quality per link and per path. The steps should be:

1. add memory to the db, so that an update does not erase previous data
2. add a cron job so that the db is updated regularly (with django-kronos)
3. plot the cost of a link with highstats
4. add a form to choose a source and destination, and plot the cost of the chosen path

**Comment by nemesisdesign** [comment]
If you want to try with this you should definitely do it in a separate module and not in django-netjsongraph itself, which aims to be lightweight, easy to install or integrate in larger projects.

The guys from wlan slovenia have implemented this with their data-stream package, I have developed a metrics module in nodeshot with influxdb.
The simplest thing you can do is to use django signals to send metrics to influxdb whenever links are updated, then you could create a grafana dashboard with the metrics and link to it from this app in someway.

**Comment by leonardomaccari** [comment]
Yes, i realized it while trying, it would complicate a lot the app. I'm working on point 1, which can be useful anyway also for django_netjsongraph, so you can visualize not only the last topology but also past ones and then separate the rest.

**Comment by nemesisdesign** [comment]
If you consider topology data to be a NetJSON blob, that also could be stored in influxdb.

**Comment by nemesisdesign** [comment]
@rohithasrk, Leonardo did some work on the snapshot feature a while ago. Take a look at his fork when you have time: https://github.com/leonardomaccari/django-netjsongraph

**Comment by rohithasrk** [comment]
Yes. @nemesisdesign : Will have a look at it.
